modules/migrations/gitea_uploader.go

```diff
@@ -690,6 +690,21 @@ func (g *GiteaLocalUploader) newPullRequest(pr *base.PullRequest) (*models.PullR
 		}
 	} else {
 		head = pr.Head.Ref
+		// Ensure the closed PR SHA still points to an existing ref
+		_, err = git.NewCommand("rev-list", "-1", pr.Head.SHA).RunInDir(g.repo.RepoPath())
+		if err != nil {
+			if pr.Head.SHA != "" {
+				log.Warn("Deprecated local head, removing : %v", pr.Head.SHA)
+				relPath := "refs/pull/" + fmt.Sprintf("%d", pr.Number) + "/head"
+				_, err = git.NewCommand("update-ref", "-d", relPath).RunInDir(g.repo.RepoPath())
```

Comment:
```suggestion
				_, err = git.NewCommand("update-ref", "--no-deref", "-d", relPath).RunInDir(g.repo.RepoPath())
```

Reply:
I added the "--no-deref". Everything works great,
here is a sample log :
`2021/09/29 10:02:58 ...dules/git/command.go:143:RunWithContext() [D] /home/********/gitea/data/gitea-repositories/99rgosse/*******.git: /usr/bin/git -c credential.helper= -c filter.lfs.required= -c filter.lfs.smudge= -c filter.lfs.clean= rev-list --quiet -1 41f9aa9fa4afdc8bb7fa1444a04801016e1d8685

2021/09/29 10:02:58 ...ns/gitea_uploader.go:697:newPullRequest() [W] Deprecated local head, removing : 41f9aa9fa4afdc8bb7fa1444a04801016e1d8685

2021/09/29 10:02:58 ...dules/git/command.go:143:RunWithContext() [D] /home/********/gitea/data/gitea-repositories/99rgosse/*******.git: /usr/bin/git -c credential.helper= -c filter.lfs.required= -c filter.lfs.smudge= -c filter.lfs.clean= update-ref --no-deref -d refs/pull/18/head

2021/09/29 10:02:58 ...ns/gitea_uploader.go:701:newPullRequest() [W] Empty reference, removing : /home/********/gitea/data/gitea-repositories/99rgosse/*******.git/refs/pull/20`

Reply:
Cool. It's much safer to use the `update-ref` command than just blindly delete the files, so I'm happy that --no-deref fixes the problem!